back-end/src/sciler/handler/handler.go

```diff
@@ -75,85 +75,67 @@ func (handler *Handler) onConnectionMsg(raw Message) {
 	contents := raw.Contents.(map[string]interface{})
 	device, ok := handler.Config.Devices[raw.DeviceID]
 	if !ok {
-		logrus.Warn("connection message received from device " + raw.DeviceID + ", which is not in the config")
+		logrus.Error("connection message received from device " + raw.DeviceID + ", which is not in the config")
 	} else {
 		logrus.Info("connection message received from: ", raw.DeviceID)
 		value, ok2 := contents["connection"]
 		if !ok2 || reflect.TypeOf(value).Kind() != reflect.Bool {
-			logrus.Warn("received improperly structured connection message from device " + raw.DeviceID)
-			return
+			logrus.Error("received improperly structured connection message from device " + raw.DeviceID)
+		} else {
+			device.Connection = value.(bool)
+			handler.Config.Devices[raw.DeviceID] = device
```

Comment:
What does this line do, since you get the device from the this map, see line 76

Reply:
handler.Config.Devices[raw.DeviceID].Connection = value.(bool) is not allowed in Go so this is the way to assign a new value to Device.Connection